arviz/plots/khatplot.py

```diff
@@ -37,45 +78,167 @@ def plot_khat(
 
     Examples
     --------
-    Plot a default khat plot
+    Plot estimated pareto shape parameters showing how many fall in each category.
 
     .. plot::
         :context: close-figs
 
         >>> import arviz as az
-        >>> centered_eight = az.load_arviz_data('centered_eight')
-        >>> pareto_k = az.loo(centered_eight, pointwise=True)['pareto_k']
-        >>> az.plot_khat(pareto_k)
+        >>> radon = az.load_arviz_data("radon")
+        >>> loo_radon = az.loo(radon, pointwise=True)
+        >>> az.plot_khat(loo_radon, show_bins=True)
+
+    Show xlabels
+
+    .. plot::
+        :context: close-figs
+
+        >>> centered_eight = az.load_arviz_data("centered_eight")
+        >>> khats = az.loo(centered_eight, pointwise=True).pareto_k
+        >>> az.plot_khat(khats, xlabels=True, annotate=True)
+
+    Use coord values to create color mapping
+
+    .. plot::
+        :context: close-figs
+
+        >>> az.plot_khat(loo_radon, color="observed_county", cmap="tab20")
+
+    Use custom color scheme
+
+    .. plot::
+        :context: close-figs
+
+        >>> counties = radon.posterior.observed_county.values
+        >>> colors = [
+        ...     "blue" if county[-1] in ("A", "N") else "green" for county in counties
+        ... ]
+        >>> az.plot_khat(loo_radon, color=colors)
 
     """
     if hlines_kwargs is None:
         hlines_kwargs = {}
+    hlines_kwargs.setdefault("linestyle", [":", "-.", "--", "-"])
+    hlines_kwargs.setdefault("alpha", 0.7)
+    hlines_kwargs.setdefault("zorder", -1)
+    hlines_kwargs.setdefault("color", "C1")
+
+    if coords is None:
+        coords = {}
+
+    if color is None:
+        color = "C0"
+
+    if isinstance(khats, np.ndarray):
+        khats = khats.flatten()
+        xlabels = False
+        legend = False
+        dims = []
+    else:
+        if isinstance(khats, ELPDData):
+            khats = khats.pareto_k
+        if not isinstance(khats, DataArray):
+            raise ValueError("Incorrect khat data input. Check the documentation")
+
+        khats = get_coords(khats, coords)
+        dims = khats.dims
+
+    if xlabels:
+        coord_labels = format_coords_as_labels(khats)
+
+    n_data_points = khats.size
 
     (figsize, ax_labelsize, _, xt_labelsize, linewidth, scaled_markersize) = _scale_fig_size(
         figsize, textsize
     )
 
     if markersize is None:
-        markersize = scaled_markersize
+        markersize = scaled_markersize ** 2  # s in scatter plot mus be markersize square
+        # for dots to have the same size
+    kwargs.setdefault("s", markersize)
+    kwargs.setdefault("marker", "+")
+
+    if isinstance(color, str):
+        if color in dims:
+            colors, color_mapping = color_from_dim(khats, color)
+            cmap_name = kwargs.pop("cmap", plt.rcParams["image.cmap"])
+            cmap = getattr(cm, cmap_name)
+            if legend:
+                msize = np.sqrt(kwargs.pop("s", markersize))
```

Comment:
It has to be removed to use Line2D to create the legend, as there, `s` is not a valid argument, it has to be `size`. This could be avoided generating the legend with `ax.scatter([], [], ...)` but unlike with `ax.plot([], ...)` I always get an error. Do you know of a way to call scatter on no data to generate a legend? It would be great

Reply:
This `s`/`size` difference is also the reason I use `markersize ** 2` instead of ArviZ default markersize from `scale_fig`. For two markers to be equal, `s` must be `size` squared.

Reply:
Okey, my bad, I was trying different things without ever changing the `c` parameter. Turns out that cmap mapping is not available for no data scatter plots, rgb tuples are though, so I will get to modify this.